web/src/app/shared/components/upgrades/master/master.component.ts

```diff
@@ -87,4 +70,8 @@ export class UpgradeMasterComponent extends BaseDirective implements DynamicComp
   cancel() {
     this.event.emit({ name: 'cancel' });
   }
+
+  needMargin() {
+    return !(this?.model?.upgrades[0]?.config?.config.length > 0 && this?.model?.upgrades[0]?.hostcomponentmap.length > 0);
```

Comment:
angular test crashed with `TypeError: Cannot read properties of undefined (reading 'length')`
I think we can add `.config?.length` and `hostcomponentmap?.length`